server/pkg/nlslayer/nlslayer.go

```diff
@@ -0,0 +1,146 @@
+package nlslayer
+
+import (
+	pl"github.com/reearth/reearth/server/pkg/layer"
+)
+
+type NLSLayer interface {
+	ID() ID
+	LayerType() string
+	Scene() SceneID
+	Title() string
+	IsVisible() bool
+	SetVisible(bool)
+	HasInfobox() bool
+	Infobox() *pl.Infobox
+	SetInfobox(*pl.Infobox)
+	Tags() *pl.TagList
+	Creator() string
+	Rename(string)
+}
+
+func ToLayerGroup(l NLSLayer) *NLSLayerGroup {
+	if lg, ok := l.(*NLSLayerGroup); ok {
+		return lg
+	}
+	return nil
+}
+
+func ToLayerGroupRef(l *NLSLayer) *NLSLayerGroup {
+	if l == nil {
+		return nil
+	}
+	l2 := *l
+	if lg, ok := l2.(*NLSLayerGroup); ok {
+		return lg
+	}
+	return nil
+}
+
+func ToLayerSimple(l NLSLayer) *NLSLayerSimple {
+	if li, ok := l.(*NLSLayerSimple); ok {
+		return li
+	}
+	return nil
+}
+
+func ToLayerSimpleRef(l *NLSLayer) *NLSLayerSimple {
+	if l == nil {
+		return nil
+	}
+	l2 := *l
+	if li, ok := l2.(*NLSLayerSimple); ok {
+		return li
+	}
+	return nil
+}
+
+type layerBase struct {
+	id        		ID
+	layerType		string
+	scene     		SceneID
+	title			string
+	visible			bool
+	infobox			*pl.Infobox
+	tags      		*pl.TagList
+	creator 		string
```

Comment:
Its better to keep the user id instead of his name, and then resolve it to the user info

Reply:
Are suggesting that creator name shouldn't be here but in another user info struct?